### ISSUE TITLE
RFC - Rearranging Symbol Libraries

### ISSUE THREAD

**Issue by SchrodingersGat** [KiCad/kicad-library#1402]
Following on from the work by @jkriege2 [here](https://github.com/KiCad/kicad-library/issues/1398#issuecomment-313969813)

| Library Name | Description | Previous Name | Notes |
| ------------------ | --------------- | --------------------- | ------------ |
| **Basic Devices** 
| device.lib | Symbols for simple devices e.g. Resistors / Capacitors / etc. | device.lib |
| elec-unifil.lib | ??? | | | 
| led.lib | LED | device.lib |
| motor.lib | Electric motors | | |
| power.lib | Power symbols (global) | | special symbols library |
| pspice.lib | Spice symbols | | |
| relay.lib | relays |
| switch.lib | Buttons and switches | | | 
| **Connectors** |
| connector.lib | Connectors, general purpose | conn.lib | Could be expanded in future e.g. *connector_usb.lib* / *connector_JST.lib* |
| **Virtual** |
| graphical_symbol.lib | | | |
| logo.lib | | |
| mechanical.lib | | | 
| **Semiconductors (simple)** |
| diode.lib | Diodes | | |
| transistor.lib | Transistors | infineon.lib | |
| thyristor.lib | Triacs, thyristors, SCR, etc | | |
| valve.lib | Valves | | |
| **Displays** |
| display_character.lib | Character displays e.g. 16x2 text display | display.lib |
| display_dot-matrix.lib |
| display_graphic.lib | Pixel-based displays, etc | display.lib |
| display_segment.lib | 7-segment, LED bar displays, etc | display.lib | |
| **Transformers** |
| transformer.lib | General purpose transformer |
| transformer_power.lib | Power transformers | 
| transformer_pulse.lib | Pulse transformers |
| transformer_signal.lib | Small signal transformers |
| **Sensors** |
| sensor.lib | Sensors that do not fit other categories | sensors.lib |
| sensor_audio.lib |
| sensor_current.lib | Current shunts / transducers / etc | sensors.lib / ir.lib / allegro.lib |
| sensor_gas.lib | Gas sensors, various | sensors.lib |
| sensor_humidity.lib | Humidity sensors | sensors.lib |
| sensor_magnetic.lib | Hall effect, etc | sensors.lib |
| sensor_motion.lib | Accelerometers / gyro | sensors.lib |
| sensor_multi-function.lib | Sensors that sense multiple things e.g. temperature / pressure | sensors.lib |
| sensor_photo.lib | Light sensors, RGB, photoresistors | sensors.lib |
| sensor_pressure.lib | | sensors.lib |
| sensor_temperature.lib | | sensors.lib |
| sensor_voltage.lib |
| sensor_touch.lib |
| **Audio / Video** |
| audio_adc_dac.lib | | digital_audio.lib |
| audio_codec.lib | Audio Codec IC | 
| audio_misc.lib | Miscellaneous audio parts |
| video.lib | | gennum.lib |
| **Linear Devices** |
| amplifier_operational.lib | Amplifiers (general purpose) | linear.lib / analog_devices.lib ||
| amplifier_audio.lib | Audio amplifiers |
| amplifier_current_sense.lib | Current sense amplifiers (separate to e.g. hall-effect current sensors) |
| amplifier_difference.lib | Difference amplifiers | linear.lib |
| amplifier_instrumentation.lib | Instrumentation amplifiers | linear.lib |
| amplifier_rf.lib | RF amplifiers |
| amplifier_video.lib | Video amplifiers
| comparator.lib | Comparators | linear.lib |
| **References** |
| reference_current.lib | Precise current reference | references.lib |
| reference_voltage.lib | Precise voltage reference | references.lib |
| **Power Management** |
| battery_management.lib | Chargers, etc | texas.lib |
| power_distribution.lib | e.g. load switch, ideal diode controller, hot-swap controllers, etc | power_management.lib, texas.lib |
| power_supervisor.lib | Reset monitor ICs, etc | power_management.lib |
| power_monitor.lib | Power monitoring devices | 
| power_protection | TVS, ESD protection, etc | |
| **Regulators** |
| regulator_module_ac-dc.lib | Modular ac-dc devices | ac-dc.lib |
| regulator_module_dc-dc.lib | Modular dc-dc devices | dc-dc.lib |
| regulator_linear.lib | Linear regulators |
| regulator_linear_controller.lib | Linear regulator controller |
| regulator_switching.lib | Switching regulators |
| regulator_switching_controller.lib | Switching regulator controller |
| regulator_switching_module.lib | Stand-alone switching modules |
| regulator_pwm_controller.lib | PWM ICs |
| regulator_pfc.lib | Power factor correction |
| **Mixed Signal** |
| analog_switch.lib | Analog switches |
| adc.lib | Analog -> Digital | adc-dac.lib |
| dac.lib | Digital -> Analog | adc-dac.lib |
| digital_pot.lib | Digital potentiometers | xicor.lib, texas.lib |
| adc_dac.lib | Devices with both ADC and DAC |
| **Drivers** |
| driver_led.lib | LED drivers | worldsemi.lib |
| driver_gate.lib | FET gate drivers, half / full bridge drivers | zetex.lib / ir.lib / power_management.lib |
| driver_motor.lib | Motor control ICs | texas.lib (?) |
| **Timers** |
| timer_oscillator.lib | Timers / oscillators / crystals | Oscillators.lib |
| timer.lib | | linear.lib | e.g. NE555 | linear.lib |
| timer_pll.lib | PLL ICs | linear.lib | e.g. NE567 | linear.lib |
| timer_rtc.lib | Real time clock | |
| **Logic** |
| logic.lib | Generic logic symbols for gates / inverters / etc |
| logic_cmos40xx.lib | 4000 series logic | cmos4000.lib / cmos_iee.lib | Previously split into IEC / IEEE variants. New library format could allow different graphical representation for each symbol, so users can choose which symbol they want |
| logic_74xx.lib | 74 series logic | |
| logic_translator.lib | Level converters, shifters, etc | texas.lib |
| isolator.lib | Optocouplers / capacitive isolation / etc | opto.lib, texas.lib |
| isolator_analog.lib | Analog isolators | | If needed |
| **CPLD / FPGA** |
| cpld_xilinx.lib | | xilinx.lib |
| cpld_altera_intel.lib | | Altera.lib |
| fpga_xilinx.lib | | xilinx.lib |
| fpga_altera_intel.lib | | Altera.lib |
| fpga_actel.lib | | actel.lib |
| fpga_atmel.lib | | atmel.lib |
| **Microcontrollers / Processors / DSP** |
| mcu_8048.lib | | |
| mcu_8051.lib | | |
| mcu_atmel_avr | Atmel AVR series | atmel.lib |
| mcu_atmel_avr32 | Atmel 32-bit AVR series | atmel.lib |
| mcu_atmel_arm.lib | Atmel ARM micro | atmel.lib |
| mcu_cypress.lib | | cypress.lib |
| mcu_cypress_arm.lib | | cypress.lib |
| mcu_microchip_pic10 | PIC10 series | microchip_pic10mcu.lib |
| mcu_microchip_pic12 | PIC12 series | microchip_pic12mcu.lib |
| mcu_microchip_pic16 | PIC16 series | microchip_pic16mcu.lib |
| mcu_microchip_pic18 | PIC18 series | microchip_pic18mcu.lib |
| mcu_microchip_pic24 | PIC24 series | microchip_pic24mcu.lib |
| mcu_microchip_pic32 | PIC32 series | microchip_pic32mcu.lib |
| mcu_freescale_hc.lib | Freescale HC series | |
| mcu_freescale_coldfire.lib | Freescale coldfire | motorola.lib |
| mcu_infineon_c166.lib | | |
| mcu_nxp_arm.lib | NXP microcontrollers | | nxp_armmcu.lib |
| mcu_parallax.lib | | |
| mcu_renesas.lib | |
| mcu_st_stm32.lib | STM32 micro | stm32.lib |
| mcu_st_stm8.lib | STM8 micro | stm8.lib |
| mcu_texas_msp340.lib | MSP340 micro | msp430.lib |
| cpu_intel_x86.lib | | |
| cpu_amd_x86.lib | | |
| cpu_motorola_68000.lib | | motorola.lib |
| cpu_motorola_powerpc.lib | | motorola.lib |
| cpu_zilog_z80.lib | | Zilog.lib |
| dsp_microchip_dspic33 | DSPIC33 series | microchip_dspic33dsc.lib |
| dsp_texas.lib | Texas DSP (digital signal processors) | dsp.lib |
| dsp_freescale.lib | Freescale DSP | dsp.lib |
| soc_texas_arm.lib | Texas ARM SOC (system on chip) | texas.lib |
| **Memory** |
| memory_card.lib | SD card, etc |
| memory_vram.lib | Volatile RAM (DRAM / SRAM) | memory.lib |
| memory_nvram.lib | Nonvolatile RAM (MRAM / FRAM) | memory.lib |
| memory_eeprom.lib | EPROM / EEPROM | memory.lib |
| memory_flash.lib | FLASH memory | memory.lib |
| memory_controller.lib | Memory controller ICs | memory.lib |
| **Interface (Comms)** |
| interface_can_lin.lib | CAN / LIN | interface.lib |
| interface_ethernet.lib | Ethernet | interface.lib / wiznet.lib |
| interface_i2c.lib | I2C interface - expanders, isolators, etc | nxp.lib, texas.lib |
| interface_lvds.lib | LVDS | interface.lib |
| interface_spi.lib | SPI interface |
| interface_telecom | Telecom interface | interface.lib, silabs.lib |
| interface_uart.lib | RS232/422/485 interface | interface.lib |
| interface_usb.lib | USB | silabs.lib,  texas.lib |
| interface_video.lib | Video interface | broktree.lib |
| **RF Interface** |
| rf_am_fm.lib | AM / FM transceivers | rfcom.lib |
| rf_bluetooth.lib | Bluetooth ICs etc | rfcom.lib |
| rf_antenna.lib | RF antenna | 
| rf_misc.lib | Miscellaneous RF parts that don't fit elsewhere |
| rf_frontend.lib | RF front-end paraphenalia |
| rf_wifi.lib | WiFi |
| rf_zigbee.lib | Zigbee / XBee |

-----------------------------------------

- ~~allegro.lib~~ - `sensor_current.lib`
- ~~analog_devices.lib~~ (AD623/AD8236/AD42x to amplifier.lib [all int amps], ADE7758 to ic_misc.lib, ADuM6000 to regulator_switching.lib, ADuM7758 to interface???.lib)
- ~~bbd.lib~~ (all audio delay lines so audio_misc.lib)
- ~~bosch.lib~~ - move to `sensor_motion.lib`
- ~~brooktre.lib~~ - move to `interface_video.lib`
- ~~contrib.lib~~ move to an `interface_<x>.lib`
- ~~ftdi.lib~~ - mostly to `interface_usb`
- ~~gennum.lib~~ - (all to video.lib: gs9020 is video processor, gs9025 is cable driver, and gs9032 is video serializer)
- graphic.lib
- ~~infineon.lib~~ (all to transistors.lib since they're just FETs with some protection built in)
- ~~intel.lib~~ - moves to multiple libs, `mcu_intel.lib`, oscillator, interface
- ~~intersil.lib~~ - moves to `driver_gate`, `driver_motor`, etc
- ~~ir.lib~~ (AU* parts to sensors_current.lib and all others to gate_driver.lib)
- ~~LEM.lib~~ - move to `sensor_current.lib`
- ~~maxim.lib~~ - split into many libs. A lot of temperature sensor parts
- ~~microchip.lib~~ - mostly interface chips
- ~~nordicsemi.lib~~ - to `interface_bluetooth`
- ~~nxp.lib~~ (all to interface_i2c.lib)
- ~~onsemi.lib~~ - `esd_protection` / `transistor`
- ~~philips.lib~~ - mostly `interface_<x>`
- ~~powerint.lib~~ (CAP* parts to ic_misc.lib and all others to regulator_switching.lib)
- RFSolutions.lib
- ~~silabs.lib~~ (cp2xxx to interface_usb.lib, si3210 to interface_telecom.lib? [it's an analog phone interface], the rest to the newly created interface_am_fm.lib)
- ~~siliconi.lib~~ (actually Siliconix parts; driver_motor.lib and analog_switch.lib)
- ~~supertex.lib~~ (reference_current.lib, power_distribution.lib, regulator_switching_converter.lib, regulator_linear.lib; see below)
- ~~texas.lib~~ (too big to write out here; see post by @evanshultz on 22 Jul below)
- ~~wiznet.lib~~ (to interface_ethernet.lib)
- ~~Worldsemi.lib~~ (to driver_led.lib)
- ~~Xicor.lib~~ (all to digital_pot.lib)
- ~~zetex.lib~~ (move to gate_drivers.lib)

### Edits

1. @schrodingersgat changed library names to singular
1. @schrodingersgat changed `microcontroller_` to `mcu_`
1. @schrodingersgat renamed `audio_adc_adc.lib` -> `audio_adc_dac.lib`
1. @schrodingersgat renamed `processor_` to `cpu_`
1. @schrodingersgat split `amplifier.lib` into multiple libraries
1. @schrodingersgat added `power_distribution`, `power_conditioning`, `battery_management`
1. @schrodingersgat added more amplifier categories, added transformer categories, improved regulator categories
1. @schrodingersgat added suggestions by @poeschlr 
1. @evanshultz divested modules.lib, gennum.lib, bbd.lib, silabs.lib, texas.lib, worldsemi.lib; noted reset ICs in power_management.lib; fixed a few typos
1. @schrodingersgat updated `power_` libraries
1. @schrodingersgat added `rf_` libraries

**Comment by SchrodingersGat** [comment]
Some notes on the (currently unassigned) libs above

1. I don't like the word "periphery" - I don't think it is technical enough
1. The remaining unassigned libs are mostly manufacturer specific. We will have to go through them and decide what functions to assign.

**Comment by SchrodingersGat** [comment]
Rather than making new tables below, please make any adjustments (after discussion) to the table in the comment at the top.

**Comment by SchrodingersGat** [comment]
## Plurals

Plural naming does not work well for everything. However I think that non-plural naming can work in all cases. Should we enforce a non-plural naming scheme is anyone really passionate about plural names?

Examples of plurals that just sound wrong

* memories_flash.lib  _(memory_flash.lib)_
* interfaces_ethernet.lib _(interface_ethernet.lib)_
* dacs.lib _(dac.lib)_

I propose we name *function_extra_sub*, otherwise we get a weird combo of *function**s**_extra_sub*, _function_extra_sub**s**_

**Comment by evanshultz** [comment]
powerint.lib has most parts that are internal FET regulators, and also a few line discharging parts that I think go into `ic_misc.lib`.

You need `audio_misc.lib` or something like that for SRCs, mic preamps, etc.

Where do CODECs go? I assume `audio_adc_dac.lib`, but what else would go there? Why not call it `audio_codec.lib`?

More libraries are needed for SMPSs more. Break out DC-DC internal FET (regulators) from DC-DC external FET (controllers). We already have a lot of each. And generic SMPS controllers like UC3842 and NCP1200 deserve their own library. They're a different ballgame than integrated DC-DC parts. Lastly, where do PFC controllers go? They're a totally different type of IC (unless you're talking about single-stage flyback or oddball topologies like the Clark converter of PFC buck) and I think they should be in their own library.

Lastly, transformers are a total wasteland. I'd like to find a paradigm that works. I've tried 3 categorization schemes and none of them worked well for me. Does anybody have anything that works? It may take multiple libraries to cover AC line frequency, SMPS, gate drive, ethernet, and all other kinds of magnetics. Additionally, do coupled inductors go here or elsewhere? What about common mode chokes? Do those have their own libraries?

**Comment by jkriege2** [comment]
Some more  comments:
- singular is fine for me
- `audio_codec.lib` is fine with me
- if we split DC-DC, then we should also have a separate lib for switched-cpacitor voltage converters (e.g. LTC660) ... name could be `regulators_switching_capacitance.lib`
- For simple magnetics like coils we don't need a separate lib (the footprints should simply use the pin-numbers 1 and 2 and they match to the generic symbol L, ...). 
- For common-mode chokes we should have a separate lib ... could we have something like `filters.lib` that could also contain other "filtering" components (most of them again fit the generic symbol, but some just might be different)??? Not sure about that
- For transformers also not sure: maybe `transformer_power.lib` (for larger ones, 220V->xyV) and `transformers_small_signal.lib` or `transformers_pulse.lib` (for audio, ethernet, ... is that the correct english name? I translated from the german "Überträger" using Wikipedia ;-)
- several libs were lost I feel (battery_management, relay_drivers.lib transistor_arrays.lib ...) I will try and add them this evening ...

Best,
JAN

**Comment by hackscribble** [comment]
+1 for singular.

There were suggested libraries starting `quarz...` that I don't see in the table above.  Would these better as `piezo...` since the material used might be crystal or ceramic?  Whether it's a filter or a resonator, it is the piezoelectric effect that they have in common.  But if we stick with the suggested name, it should be `quartz...` (I'm being very picky :wink:)

What will be left in `device` if inductors move to a separate library as @jkriege2 suggests above?  Just resistors and capacitors?  If so, why not split into two libraries for clarity?

How about shortening `microcontroller_` to `mcu_`? Anyone using one ought to recognise the acronym.

`peripheral` instead of `periphery`?

**Comment by SchrodingersGat** [comment]
1. Alright, singular names it is (I have updated the table accordingly).
1. I also prefer `mcu_` as a prefix (we already have `soc_` and `dsp_`
1. Regulators (switching). Agreed that we need to arrive at some categorizations here
1. Transformers - my knowledge here is lacking
1. `peripheral.lib` - the suggested contents of this library still seem very broad to me. Can we narrow this down?

**Comment by hackscribble** [comment]
> piezo / quart / etc - what is the function of this library (not the material it is made of!) Is it oscillators? Resonators? Other?

OK, I see that there have been edits on the other thread that prompted my comment :smile: : https://github.com/KiCad/kicad-library/issues/1398#issuecomment-313926433

That now suggests two libraries:

> oscillators.lib (formerly: Oscillators.lib)
quarzes_filters.lib (formerly: Oscillators.lib)

If `oscillators.lib` is passive piezoelectric oscillating devices (i.e. standalone crystals, crystal resonators with built-in capacitors and ceramic resonators), that's good.  But it should not IMHO include oscillator ICs.

Piezoelectric filters can be made of quartz crystal or ceramic material.  So `quartz_filters.lib` does not sit well with me.

**Comment by pointhi** [comment]
After a quick look
* ```audio_adc_adc.lib``` -> ```audio_adc_dac.lib```
* ```sensor_multiple.lib``` -> ```sensor_combined.lib``` is probably a better name. Probably someone has a even better one

@SchrodingersGat probably ```kicad-symbols``` would be a more appropiate name for the repo than ```symbols```

**Comment by nbxmike** [comment]
You may as well combine the NXP and Freescale processors to NXP or you will need to do it later.

**Comment by evanshultz** [comment]
@pointhi Can you please check you first bullet? It looks like the same lib name is repeated.

Unless someone objects, I volunteer to propose the divesting of the following libs:
- analog_devices
- infineon (which purchased IR)
- intersil
- on semi
-zetex

**Comment by evanshultz** [comment]
@SchrodingersGat Is the above a "all edit" table? Or are we making suggestion with one person (you, I assume) as mediator and editor of the table?

**Comment by evanshultz** [comment]
Finished with analog_devices.lib, infineon.lib, and zetex.lib.

I like the idea of separate diode libraries IF the symbols are different. For example, regular Si diodes, Schottky/SiC, zener/TVS, etc. This will result in the least amount of unique symbols so it's the easiest to maintain.

**Comment by diggit** [comment]
In _my_ opinion, there is no need for symbols of specific symbols for such simple devices. Generic symbols are IMO enough. There is only several permutations with so few pins. Then splitting them does not make sense, they could all got to device.lib.
At least my workflow is:

1. find online which components suits the purpose and is available
2. choose correct generic symbol (it is fast, keywords like: gds, zener, tvs, diode, fet, npn, .... works well)
3. rename symbol to match selected device
4. assign correct FP

I have similar workflow for linear regulators, because in 70% cases, the one I choose to use is not in libs.

**Comment by SchrodingersGat** [comment]
> @evanshultz - @pointhi Can you please check you first bullet? It looks like the same lib name is repeated.

Read it again ;) It's a very subtle mistake that I made in the original table, which I have now fixed.

**Comment by SchrodingersGat** [comment]
> @SchrodingersGat Is the above a "all edit" table? Or are we making suggestion with one person (you, I assume) as mediator and editor of the table?

It's open to editing, but only after discussion here :) Please add edit comments at the end of the table as I have been doing, if you edit anything!

**Comment by SchrodingersGat** [comment]
### Amplifiers

Currently the amplifier organization is pretty rough. What are some broad categorizations for amps?

* amplifier_instrumentation
* amplifier_current_sense
* amplifier_chopper
* amplifier (everything else)

**Comment by gauravjuvekar** [comment]
@SchrodingersGat Er, I don't think others can edit the post. That's not how github works.

**Comment by SchrodingersGat** [comment]
Sorry @gauravjuvekar you are completely correct - please ignore that.

Update : Make suggestions and I will add them to the table :)

**Comment by jkriege2** [comment]
I think you need to be Member or Owner to edit a non-own post.
JAN

**Comment by gauravjuvekar** [comment]
Can we create separate `rf_*` libs. I know that currently there aren't many rf components in the libs, but there are a large number of categories in rf out there.

Eg, see https://www.digikey.com/products/rf-wireless/en
I guess rf_{trasmitters, receivers, transceivers} could go in `interface_rf`, but there are a lot of non-interface components like amplifiers, mixers, couplers, active balun ICs, switches, etc.

**Comment by SchrodingersGat** [comment]
@gauravjuvekar this is a good idea, I was only just thinking where antennae would go. I agree there is a place for `rf_` libs separate from RF ICs

**Comment by gauravjuvekar** [comment]
I think antenna should also be a separate library. It also has a lot of subdivisions (type: patch, chip, wire)

**Comment by jkriege2** [comment]
@SchrodingersGat: some comments about amplifiers:
- `amplifier_current_sense` shouldn't/couldn't that go to `sensor_current`?
- I'm missing `amplifier_difference` (for components like these: http://www.ti.com/lit/ds/symlink/ina157.pdf) ... basically OPAMPS, but with (calibrated) resistors inside
- Also maybe `amplifier_power` and/or `amplifier_highvoltage`? (not so sure about these ... but there we could put e.g. opamps that also work at 100V ...)

Just ideas ...

JAN

**Comment by SchrodingersGat** [comment]
> amplifier_current_sense shouldn't/couldn't that go to sensor_current?
I think that's fair.

> I'm missing amplifier_difference (for components like these: http://www.ti.com/lit/ds/symlink/ina157.pdf) ... basically OPAMPS, but with (calibrated) resistors inside
> Also maybe amplifier_power and/or amplifier_highvoltage? (not so sure about these ... but there we could put e.g. opamps that also work at 100V ...)

I've added `_difference`, `_power`, and `_rf` but I don't think that `_highvoltage` is a category all on its own...

**Comment by SchrodingersGat** [comment]
Added transformer categories.

**Comment by jkriege2** [comment]
will people know that `amplifiers.lib` or `ic_amplifiers.lib` contains OPAMPS? ... i.e. are all amplifiers OPAMPS or one of the others ...

JAN

**Comment by poeschlr** [comment]
What about (multi purpose) signal level translators and signal isolators? (currently they might go into the logic libs but i think giving them a separate lib might be a good idea.)

The symbols of the LEM lib could go into the sensor_current lib.

**Comment by evanshultz** [comment]
> amplifier_current_sense shouldn't/couldn't that go to sensor_current?

I disagree. The sensor is the part that actually senses the current. ICs like we see in `linear.lib` now that are "current sense" parts are simply amplifying an external sensor, like a resistor or transformer/inductor winding. I do not think those ICs belong in `sensor_current.lib`.

I also do not like having so many sub-categories for opamps. Why would you break out chopper-based amps? Their benefit is low input offset voltage, but you can get non-chopper amps with excellent performance here as well. It seems that we'd be splitting up parts which are otherwise interchangeable, cutting users off from seeing all opamp possibilities they might want to choose.

**Comment by poeschlr** [comment]
Gennum  is now semtech
the `gennum.lib` would need to be split up into multiple libs. (non of the symbols have a description which makes this a bit harder.)

`graphic.lib` does not exist in the current master branch. It has been renamed to `graphic_symbols.lib` (or has it been deleted and the graphic_symbols lib is a completely new lib?)

`intel.lib` mostly into a new `mcu_intel.lib` but also some into `oscillator`, `interface*`, ...

`intersil.lib` either `driver_gate` or `driver_motor`

`maxim.lib` split up in many parts. (many temp sensors but also a some other stuff.)

`microchip.lib` mostly interface chips. (some gate drivers)

`nordicsemi.lib` only one chip is in there. should go to `interface_bluetooth`

`nxp.lib` some i2c led controllers and i2c multplexers

`onsemi.lib` mostly `esd_protection` but also `transistor`

`philips.lib` mostly interfaces

---

Where should digital potis go?
Where should i2c, spi, ... io expander go?
should h-bridge drivers go into `diver_gate` or `driver_motor`?

**Comment by nbxmike** [comment]
At the rate at which sub-categories are being proposed, the library revision better have some kind of search tool across ALL libraries.  

I am no op-amp expert and rely on others to tell me that part of a design, so I would not necessarily know which of the categories to look in for a part number I've been provided.  I doubt I am alone in allowing other people with more specific expertise to dictate portions of my designs, I could see us getting very frustrated with so many finely sliced parts libraries.  Right now, grep is my go to solution for this but I think less command line experienced people would prefer a built in tool.

**Comment by poeschlr** [comment]
When placing a part in eeschema the search field gives results in all libs.

**Comment by poeschlr** [comment]
The note for connector lib:
> Could be expaned in future e.g. connector_usb.lib / connectos_JST.lib

I don't think generic connectors need a specialized symbol.
The only symbols we could add is for multi row connectors where each row has a separate connector on the wire side. Or for connectors that don't follow a pure numeric numbering scheme.

**Comment by jkriege2** [comment]
@poeschlr 
I would say:
- dig. potis in `ic_digpotentiometers` or `digital_potentiometers` ... there are so many that I think they deserve their own lib ... anotherpossibility (although not completely accurate would be `dac.lib`)
- h-bridges I would put into driver_motor?
- for IO-expanders, RTC, ... I proposed the periphery_-tree ... which was not accepted ... as RTCs are in `rtc.lib`, this would call for `io_expanders.lib` ... 

@SchrodingersGat :
- what about timers like 8253, which are specifically designed as µC-periphery ... do they fit in `timer.lib`?

JAN

**Comment by jkriege2** [comment]
About OPAMPs: I wouldn't separate out the chopper-amps ... as far as I know they are (to the outside) like OPAMPs (they have In+, In- and VoltageOut .. possibly+control pins) ... but OTA-amplifiers, or transconductance amplifiers have a current output, not a voltage output ... shouldn't they be separated out?

**Comment by SchrodingersGat** [comment]
@poeschlr I have applied your suggestions to the table, thanks :)

@jkriege2  I have added `digital_potentiometer.lib` :) I think `driver_motor` contains standalone motor controller ICs, while `driver_gate` contains things like half/full-bridge drivers etc (they're used for more than just motors)

**Comment by SchrodingersGat** [comment]
> @jkriege2 what about timers like 8253, which are specifically designed as µC-periphery ... do they fit in timer.lib?

Yes, I'd expect so.

**Comment by jkriege2** [comment]
@SchrodingersGat about drivers: Just starting from the lib-name in `driver_gate` I would expect driver-ICs that are used to drive the gate of a MOSFET, not half or full bridge drivers (although both might share the same functioning principle ...) ...

I scanned a bit the webpages of semiconductor manufacturers:

- http://www.ti.com/lsds/ti/analog/motordrivers/overview.page
- https://www.infineon.com/cms/de/product/power/channel.html?channel=db3a304319c6f18c011a154646852706
- http://www.nxp.com/products/automotive-products/smart-power-drivers:SMART-POWER-DRIVERS
- http://www.onsemi.com/PowerSolutions/taxonomy.do?id=210
- http://www.st.com/en/motor-drivers.html

Most of them separate (at least) between:
1. MOSFET/IGBT drivers
2. Motor drivers/controllers (with sub-cats like stepper, bldc, DC, ...)
3. some also have load/relay drivers

I think we should use the same categories, e.g.
1. `driver_motor` (for all motor-related products)
2. `driver_gate` (for ICs that are supposed to drive high-power MOSFETs/IGBTs)
3. `driver_load_other` or `driver_load` for all others
4. maybe we could think about having `motor_controller` for motor-controll ICs like stepper-controllers or BLDC-controllers that do not necessarily contain drivers for the motor.

**Comment by jkriege2** [comment]
Also I just stumbled about MCU-companion chips like RESET-ICs and voltage supervisors ... where to put those? `periphery`?

What about the IO-expanders and other periphery that does not fit in any existing lib (yet)?

JAN

**Comment by SchrodingersGat** [comment]
Reset ICs should go in `power_supervisor.lib` - this lines up with how they are categorized on [digikey](https://www.digikey.com/products/en/integrated-circuits-ics/pmic-supervisors/691).

I/O expanders (etc) are yet to be allocated. As no better name has been suggested, perhaps a series of `peripheral_` libraries is needed.

**Comment by jkriege2** [comment]
+1 on both ;-)

**Comment by jkriege2** [comment]
- if IO-expanders are the only ones left ... maybe `interface_parallel.lib` would also be an option? + then we don't have to open up another "tree"
- We could still have a lib `periphery.lib` for everything that does not fit in any other category ...
- Also I think we don't have any place for I2C transceivers/drivers/hubs and SPI switches/transceivers/... maybe `interface_spi.lib` + `interface_i2c.lib`

**Comment by gauravjuvekar** [comment]
most of `ftdi` will also go to `interface_usb` or `interface_rs232`

**Comment by poeschlr** [comment]
`opto_isolator.lib` might not be the best name. (And opto maybe not the best category)
Opto is more the how is an isolator implemented than a functional description.

Maybe better would be `isolator_opto` and further categories for each type of isolator 'isolator_induction' 'isolator_capacitiv'. 
But i don't like categorizing by internal technology. Yes inductive isolators tend to be used for signals with higher frequencies but there might be opto-isolators out there that can handle similar frequencies to cheap induction based isolators. (or there are specialized induction based isolators for low frequency/ low power applications)

I would not split them up to be honest. maybe just `isolator_digital`? (I don't think there are analog isolators. If they are ever invented or if a contributor finds one we can simply add a new lib called isolator_analog)

Another lib still missing is the `level translator`. 

I personally would place the level translator and the isolator stuff either under logic or under their own signal translation category. (The later might be more future proof.)

**Comment by poeschlr** [comment]
Something similar with the display libs.
under `display_lcd` there will be monchrome and color displays. Under display led i think there mitht be led bars, 7 segment, oled displays and in future maybe full color displays (TVs use leds already maybe in future there will not be lcds. also lcd display has a redundant display in it. it should be display_lc).

i would suggest the following function based lib names:
- display_bar (or display_led-bar but this would again be technology based not function based)
- display_segmented (7, 9, 14 and 16 - segment displays) 
- diplay_dot-matrix
- display_monochrom
- display_color

**Comment by SchrodingersGat** [comment]
@poeschlr regarding the `display_` libraries:

I think there are few enough symbols that `display_led` and `display_lcd` are OK (at least for now). We have to ensure that we don't over-specialize, creating too many libs.

If a certain library becomes too large we can always split further down the road.

**Comment by SchrodingersGat** [comment]
> opto_isolator.lib might not be the best name. (And opto maybe not the best category)
> Opto is more the how is an isolator implemented than a functional description.

Fair point. What about just `isolator.lib` - if we find this has too many symbols we can later expand by sub-category.

**Comment by SchrodingersGat** [comment]
> Also I think we don't have any place for I2C transceivers/drivers/hubs and SPI switches/transceivers/... maybe interface_spi.lib + interface_i2c.lib

@jkriege2 can you give me some examples of what you mean by such components? :) It will help me to better understand their categorization.

**Comment by SchrodingersGat** [comment]
> I personally would place the level translator and the isolator stuff either under logic or under their own signal translation category. (The later might be more future proof.)

Should we add `logic_translator.lib` for e.g. parts that are not covered by the 40xx and 74xx series?

**Comment by poeschlr** [comment]
@SchrodingersGat wrote:
> I think there are few enough symbols that display_led and display_lcd are OK (at least for now). We have to ensure that we don't over-specialize, creating too many libs.

I really think led and lcd are bad names. there are lcd 14 segment displays.
If you only want to libs maybe the following might be a good split:
- `display_graphical` (Pixel based. Pixels spaced evenly across the display area. Can later be split into `display_graphical_monocgrom` and `display_graphical_color`)
- `display_simple` (segment based, dot matrix, bars, ... Can later be split into `display_bar,` `display_segmented`, `display_dot-matrix`)

**Comment by SchrodingersGat** [comment]
How about `display_matrix` (pixel based displays), `display_segment` (bar, 7-seg, etc) then?

**Comment by poeschlr** [comment]
dot-matrix displays fall (for me) more in the same category as 7 segment (or better 14 segment) displays because the have a clear separation between there "letters"

I think "graphical" describes the capabilities better. Simple might not be the best name. But would you search dot matrix displays or bar displays under `display_segment`?

**Comment by SchrodingersGat** [comment]
Should we just leave it as `display.lib` and then see where the categorization falls as the library grows?

**Comment by poeschlr** [comment]
TlDr: I would split the display libs along the line of graphical/non graphical

I would make it a bit more future proof. I would aim for something where we only need to split libs again when v6 comes out. (In other words, if we think this lib might be split as soon as more symbols are added we should split it now.)
If @Misca1234 carries on as he does now we will shortly reach that point. (one display symbol, footprint 3d model a week -> that sums up over the next 2 to 3 years until v6 comes out.)

But i might be wrong. Your call. It is hard to predict where the contributors take us. (And i'm sure we will be forced to split a few libs during the lifetime of v5 while others stay "empty". I just hope to limit this as far as possible.)
I understand why limiting libs might be a good idea. (Some people would like one large lib. I'm more the fine grained type. I don't think from a maintainance point of few there are problems as long as it is easy to see where a symbol should go. There is of course the danger that there exists a component that falls into multiple categories.)

---

Side note, (explains my stance a bit i think):
I hope that the contribution frequency stays at the current rate or even increases further. (The last year saw much more activity then the year before that. That year saw more activity than the year before, ...)
A large part of that is a reduced response time between initial PR opening and first feedback (even if it is only by travis) And i think the clear KLC also helps. A clear library structure might further increase interest by contributors. (Or everybody hates us for putting the bar too high and they stop contributing. God i hope not.)

**Comment by SchrodingersGat** [comment]
@poeschlr I appreciate your points - if we can predict future use it is better to make the change now rather than later. I'll ponder over this one (it's way too late at night here already :) )

**Comment by jkriege2** [comment]
@poeschlr There actually are analog isolators (usually called isolation amplifiers, see e.g. http://www.analog.com/en/products/amplifiers/isolation-amplifiers/ad215.html ... and we even have some in the LIB) ... and there are parts that combine an LED and an LDR, which can also be used as an analog isolator (although slow).

**Comment by jkriege2** [comment]
@SchrodingersGat here are some I2C examples:
- http://www.ti.com/lsds/ti_de/interface/i2c-multiplexer-switch-products.page
- http://www.ti.com/lit/ds/symlink/iso1541.pdf
- http://www.ti.com/lsds/ti_de/interface/i2c-hub-buffer-repeater-products.page, http://www.ti.com/lit/ds/symlink/pca9518.pdf
- http://www.ti.com/lit/ds/symlink/pca9515a.pdf
- http://www.nxp.com/products/interfaces/ic-bus-portfolio/ic-bus-repeaters-hubs-extenders:MC_41849
- http://www.nxp.com/docs/en/data-sheet/P82B715.pdf

and for SPI:
- http://www.ti.com/lit/an/slyt441/slyt441.pdf

**Comment by jkriege2** [comment]
@SchrodingersGat What about the name `logiclevel_translator.lib` ... level is maybe a bit generic?

**Comment by jkriege2** [comment]
About displays: I would like to see
- `display_character` (7seg, typical LCD-displays for text ...)
- `display_graphic` (graphic displays maybe also LED DOT-matrix in here)

One could think about splitting out the two special cases:
- `display_dotmatrix` (dot-matrix, bars, ...)
- `display_segmented` (7-segment, ...)

but not really sure ... what is best

Best,
JAN

**Comment by SchrodingersGat** [comment]
Ok I have updated `graphic_` and added `interface_i2c` and `interface_spi`. Time for bed :) Thanks for the suggestions.

**Comment by poeschlr** [comment]
> About displays: I would like to see
    display_character (7seg, typical LCD-displays for text ...)
    display_graphic (graphic displays maybe also LED DOT-matrix in here)

I like that. My only problem is where do led bar displays fit here. Maybe they should go into the led lib? (Can be seen as led array. Has similarities to a resistor array.)

**Comment by poeschlr** [comment]
What about `relay.lib`
Maybe `relay_mechanical` and `relay_solid_state`? (The later is currently often in opto.lib, SSR are not really used for the same usecases as isolators.)

**Comment by poeschlr** [comment]
What is the difference between `regulator_switching_module.lib` and `power_module_dc-dc.lib` or `power_module_ac-dc.lib` (I think this came out of a suggestion by @evanshultz but i can't really follow it.)

**Comment by jkriege2** [comment]
More questions:
1. Should we split `transistor.lib` maybe into `transistor_bipolar.lib`, `transistor_fet.lib`,  `transistor_igbt.lib`,  `transistor_ujt.lib`
2. General question: Should the generic transistor/diode/... symbols from `device.lib` reside there, or should they be moved to the respective specialized libs? Maybe not, as they are no good bases for ALIASes, as they don't specify packages.

Best,
JAN

**Comment by evanshultz** [comment]
I still feel there's a level of classification missing. Libraries with ICs in them should start with `ic_`, such as `ic_comparator.lib`. Otherwise we will have libraries all spread out that are related. And it further clarifies the content of a library, so that it's clear to users that ICs are in this library.

Why not rename `triac` to `thyristor`? The explanation says it's for all thyristors, and these are pretty rare so lumping them all together should be fine. If not, it seems odd to put diacs, scrs, and other thyristors into a library named `triac`?

Can we fill in the expected use of the `transformer` libraries? We would have power transformers at the line frequency, high frequency power transformers (SMPS), gate drive transformers, audio bandwidth transformers (both high power and low power), ethernet/digital transformers, etc. How do all of those map to the libraries above? And also, how do we name transformers? If the transformer is a single-source part then we could use the MPN. But what if it's a custom or generic transformer?

I think we need `audio_misc.lib` or `ic_audio_misc.lib`. Where do SRCs go? DIRs and DITs? Audio volume control ICs? Mic preamps like THAT1583 and THAT5263. Those are parts in the market, some of them are in the library now, and it's unclear to me where they should live.

Where does a part like AD7533 go? This is an existing library part.

I would like to change `amplifier.lib` to `amplifier_operational.lib` or `amplifier_opamp.lib` (but that sounds weird). Otherwise it's ambiguous to me what goes into that library. Audio power amplifiers would seem to fit. What about OTA? It's better than `linear.lib`, but it's still not well-classified to me.

Does `amplifier_audio.lib` encompass linear and switching power amplifiers? Like TPA3251 and TDA7577 and MAX9710 and MAX9715? And why is it not in the "Audio / Video" section?

What exactly is in `power_conditioning.lib`? Is this some AC in, DC out PFC pre-regulator?

Here is your answer, @poeschlr. `power_module*` libraries are complete, OTS power supplies. They do come in both varieties (AC in and DC in, with DC out) and I think a user would reasonably expect them to be segregated. That part is fine. I have no idea about `regulator_switching_module.lib`. However, since we have `regulator_switching_controller.lib` we should have `regulator_switching_converter.lib`. These are external switch (not only FET) and internal switch parts, and they are grouped differently. Check out the part taxonomy at any IC vendor or distributor. But there is another group of parts: generic PWM controllers like SG3525 and TL494. Those could be separate from the parts in the libraries above which are generally relatively fixed-function DC-DC converters. So I think that's a third library, perhaps named `regulator_smps_controller.lib` or `regulator_pwm_controller.lib`. PFC controllers deserve, I think, their own library. Suggest `regulator_pfc_controller.lib`. Is that too much?

Should `adc-dac.lib` be `adc_dac.lib` (note underscore?

I assume `driver_gate.lib` here is only for ICs since transformers appear to be captured above. Another reason to preface IC libs with `ic_`. Would this lib include audio-focused gate drivers like IRS20957 and IRS2092? Those are certainly not a general-purpose gate driver and we have broken out audio parts for other categories.

Why `timer_oscillator.lib` if it only contains oscillators and crystals? Wouldn't `crystal_oscillator.lib` or `clock.lib` be a better name?

Since there's a lot of overlap, where many ICs will handle both protocols, how about `interface_rs232_rs485.lib` to combine the two separate libs?

To @jkriege2 
1. I think they're close enough to leave together. We'd need `transistor_igbt.lib` and a ton of others. Lots of library management overhead for probably little benefit, from my perspective, since in many circuits I could use either. Would having `transistor_array.lib` (for parts with multiple transistors) fall into the same category of being too much?
2. I like this idea. At least if there's a symbol update we don't have to hunt around in multiple libraries. Seems easier to manage if like parts (and symbols) are kept together.

**Comment by poeschlr** [comment]
If we mark libs that have ics in them with ic_ we have 80% of the libs starting with ic_ Not sure if that helps anyone. 
I see ic more as a technology than a function.

**Comment by SchrodingersGat** [comment]
> @jkriege2  - General question: Should the generic transistor/diode/... symbols from device.lib reside there, or should they be moved to the respective specialized libs? Maybe not, as they are no good bases for ALIASes, as they don't specify packages.

My thinking was the `device.lib` contains generic transistor symbols without package association. `transistor_x.lib` is for specific parts with footprints assigned.

> @poeschlr  - If we mark libs that have ics in them with ic_ we have 80% of the libs starting with ic_ Not sure if that helps anyone.
> I see ic more as a technology than a function.

Definitely agreed. `ic_` is not the important functional qualification, and MOST devices will be ICs anyway.

> @evanshultz - why not rename triac to thyristor? 
> I think we need `audio_misc.lib` 

Ok, done! :+1: 

> @evanshultz - what exactly is in power_conditioning.lib? Is this some AC in, DC out PFC pre-regulator?

I was thinking PFC etc

> Since there's a lot of overlap, where many ICs will handle both protocols, how about interface_rs232_rs485.lib to combine the two separate libs?

I have merged all RS232/422/485 into `interface_uart.lib`

> @evanshultz  - Why timer_oscillator.lib if it only contains oscillators and crystals? Wouldn't crystal_oscillator.lib or clock.lib be a better name?

Fair point, I see crystals and clock ICs as being similar but separate. Perhaps two separate libs?

**Comment by poeschlr** [comment]
General question: can we reduce the amount of acronyms here? I'm getting lost a bit. (This might be the reason why i still don't fully understand the description by @evanshultz. What is an OTS power supply?)

I think we should keep this discussion as clear as possible to help us and future contributors determine what symbol should go in what lib. (Remember: not everyone learned this stuff in english)

**Comment by SchrodingersGat** [comment]
Question - where should things like SD-cards / SIM card live?

**Comment by evanshultz** [comment]
> I was thinking PFC etc

@SchrodingersGat I still don't know what you mean. Can you give an example?

@poeschlr My apologies. OTS = off the shelf. Not custom-designed. My understanding is that if the lib name includes `module` it would be for something like [this](http://www.meanwellusa.com/webapp/product/search.aspx?prod=EPS-15). I don't know why there would be 3 `module` libs, though.

**Comment by evanshultz** [comment]
on_semi.lib divestiture:
- cm1213a -> esd_protection
- ncp45560* is a load switch -> `transistors.lib`? (these are like the load switches from `infineon.lib`)
- nupx202 -> esd_protection

**Comment by evanshultz** [comment]
All parts in `allegro.lib` can go to `sensor_current.lib`.

**Comment by evanshultz** [comment]
nxp.lib mapping:
- pca9536* -> `interface_i2c.lib`
- pca9544* -> also `interface_i2c.lib` I guess
- pca9685* -> interface_i2c.lib

**Comment by evanshultz** [comment]
All `xicor.lib` can go to `digital_pot.lib`.

**Comment by evanshultz** [comment]
`Worldsemi.lib` (which doesn't need to be capital) are all integrated/smart LEDs. They don't fit into `display*.lib` or `driver_led.lib`, I don't think, and `led.lib` is generic. So... anybody have ideas?

**Comment by poeschlr** [comment]
Well we put the smart transitors made by infineon into the transistor lib. By that logic smart leds can go into the led lib.

**Comment by evanshultz** [comment]
supertex.lib break out:
- cl220 -> `reference_current.lib` (it's a general-purpose current regulator, even though they mention LEDs)
- hv10x -> `power_distribution.lib`
- hv992* -> `regulator_switching_converter.lib`
- hv9930/hv9960/hv9967/hv9972 -> this goes with generic PWM controllers
- hv9931 -> this goes with PFC controllers
- lr8* -> `regulator_linear.lib`

**Comment by evanshultz** [comment]
The `siliconi.lib` should really have been `siliconix.lib`, and the parts inside should go:
- d469 -> `driver_motor.lib` (seems like the best option to me)
- dg411 -> `analog_switch.lib`

**Comment by evanshultz** [comment]
That's a good chunk of the libraries, but texas.lib is huge and I didn't touch it. I'll tackle that one tomorrow if nobody else has done it.

**Comment by poeschlr** [comment]
I would not really list exactly which symbol will go where. A better way is to find out which libs will be needed in the new lib system for splitting the legacy lib. What symbol goes where is better determined while moving a lib to the new system. (otherwise we do double the work.)

**Comment by evanshultz** [comment]
They will need to move at some point, and I mentioned if there wasn't an obvious mapping to the libraries already created above. Isn't that the same goal?

**Comment by jkriege2** [comment]
Hi!

the analog multipliers are also a problem ... maybe we need something like `analog_misc.lib` or `linear_misc.lib` for such special function ICs or anything that doesn't fit anywhere else ...

JAN

**Comment by SchrodingersGat** [comment]
@jkriege2 such libraries will become apparent as we reorganize and find symbols that need a home but don't have one. 

> Question - where should things like SD-cards / SIM card live?

Any ideas on this one?

**Comment by poeschlr** [comment]
or a new lib under the connectors category? example: `connector_card` (maybe a better name is needed.)

**Comment by jkriege2** [comment]
iff we split `connectros.lib` ... Has there been a consensus on that?

**Comment by suzizecat** [comment]
> About SD/SIM: Why not keep them in connectors? They are "just" connectors for such devices in the end ...

Even if finally, it doesn't change anything, there is some "eSIM" card which should become available soon. These are SIM card which you'll solder directly on your PCB, so these won't be "just" a connector for a long time... I would vote for the `connector_card` or even only `card` since you'll be able to add some other cards readers (like smart cards). 

It's just a suggestion thought since i absolutely don't know how to handle these other cards...

**Comment by pointhi** [comment]
what about simply adding a ```connectors_misc.lib``` for things like sd-card-connectors?

**Comment by stevenj** [comment]
This all looks great, my only comment so far is:

mcu.lib - Micros that don't fit anywhere else
cpu.lib - Processors that don't fit anywhere else
dsp.lib - DSP chips that don't fit anywhere else
soc.lib - SOC chips that don't fit anywhere else

Whats actually in these now?
It seems like future maintenance rot waiting to happen. 
Ie, If a Silicon Labs EFM32 gets contributed, it doesnt have anywhere to go, because there isnt a silabs_mcu_arm library.  And its only one chip, so meh just stick it in mcu.lib

But then another gets contributed and so on,  I think it would be better NOT to have generic mcu/cpu/dsp/soc libs and make specific libs for whatever is in there, even if that means they only have one part.  Then in future if someone contributes a part, they dont have a dumping ground to stick it, and the process is to create a new library following the scheme already laid out.

**Comment by poeschlr** [comment]
connector_misc only makes sense if we have specific connector libs for different connectors. (Which we don't really have)
Currently the connector lib is the connector_misc lib.

**Comment by evanshultz** [comment]
Just a few part types to consider. I realize they all aren't in the library now, but where would we classify them?
- vactrols like [VTL5C2](http://www.alliedelec.com/m/d/ecb1ed91959ae54d97cf67f5103d9b2b.pdf)
- RMS-to-DC converter like [AD736](http://www.analog.com/media/en/technical-documentation/data-sheets/AD736.pdf)
- V-to-F converter like [AD654](http://www.analog.com/media/en/technical-documentation/data-sheets/AD654.pdf)
- DDS generator like [AD9852](http://www.analog.com/media/en/technical-documentation/data-sheets/AD9852.pdf)

I'm not really sure about the vactrol, but the rest may have no other place but a miscellaneous IC library which I still don't see above?

**Comment by poeschlr** [comment]
google tells me vactrols are isolators. Which would mean they go into isolator.lib? (or isolator_analog.lib)

**Comment by SchrodingersGat** [comment]
@evanshultz good points - I'm not sure immediately where they should go.

**Comment by evanshultz** [comment]
I still think the regulator library names need some work. I was finally able to sort it out in my head.

The word "regulator" is used for the first term in these libraries. But it's also common to differentiate DC-DC ICs with the words "regulator" if the transistor (not necessarily a FET) is inside and "converter" if the transistor is external. That confluence of terms is what was bothering me. I think the first word in the library should change to "power", and then it's not awkward to use the name "regulator" elsewhere in the library name.

With that sorted out, I propose the following:
- `power_linear_regulator`: integrated linear reg like 7805 (as above)
- `power_linear_controller`: linear reg IC that uses an external switch like LM723 (as above)
- `power_switching_regulator`: predominantly DC-DC IC with internal switches like LM2574
- `power_switching_controller`: predominantly DC-DC IC with external switches like TPS40200
- `power_pwm_controller`: generic PWM ICs like UC3842
- `power_pfc_controller`: PWM ICs for PFC supplies, like LT1248

I scrolled up and realized there already was a Power Management section using `power` as a prefix. Regardless of what section the above goes into, I still think this is a better way to go. Also, `power_switching_module` could still be retained.

I went around the web to see how others had classified these types of parts, both as manufacturers and as sellers/organizers of these parts.

Here are Digi-Key's categories:
![image](https://user-images.githubusercontent.com/1936989/28469022-c1ae884e-6de9-11e7-9272-4db0e59c665f.png)

And Mouser's:
![image](https://user-images.githubusercontent.com/1936989/28469065-e5f842c6-6de9-11e7-8de2-64055bbd8c7c.png)
![image](https://user-images.githubusercontent.com/1936989/28469088-f3f4bb70-6de9-11e7-94e1-8fba67dd626f.png)

Farnell:
![image](https://user-images.githubusercontent.com/1936989/28469155-36a8bc14-6dea-11e7-838a-0e101eda4ac6.png)

TI:
![image](https://user-images.githubusercontent.com/1936989/28469201-60b7ef5c-6dea-11e7-8898-d2897e10bb6a.png)
![image](https://user-images.githubusercontent.com/1936989/28469218-750fde1a-6dea-11e7-980f-02b5788b470f.png)

Linear Tech:
![image](https://user-images.githubusercontent.com/1936989/28469305-b7410386-6dea-11e7-988d-ffa79679f2ca.png)

ST:
![image](https://user-images.githubusercontent.com/1936989/28469428-0a40e84e-6deb-11e7-9e7d-30cf55123828.png)

**Comment by evanshultz** [comment]
Breaking out the last of the old libraries (that I see haven't been divested):
- `bbd.lib`: all audio delay so I vote for `audio_misc.lib`
- `gennum.lib`: all video ICs that should go to `video.lib`
- `modules.lib`: all MCU board shields; can they be turned into templates?
- `silabs.lib`: cp2xxx to `interface_usb.lib`, si3210 to `interface_telecom.lib`? [it's an analog phone interface], the rest to the newly created `interface_am_fm.lib`
- `worldsemi.lib`: to `driver_led.lib`

`texas.lib` is huge so I'm giving it a section to itself (and using the names I proposed above):
- DRV8860*: without a generic driver library, I'd go with `driver_motor.lib`
- ISO*: `isolator.lib`
- LM3*/LM4*/TM4*/TMS470*: `soc_texas_arm.lib`
- LM5175: `power_switching_controller.lib`
- LP3947: `battery_management.lib`
- pca9306: `interface_i2c.lib`
- tca9555: `interface_i2c.lib`
- tlc555: remove as I already have a fixed-up version in `linear.lib` at https://github.com/KiCad/kicad-library/pull/1081
- TLC5940/TLC5973: `driver_led.lib`
- TPL0401: `digtal_pot.lib`
- TPL9201: torn between `driver_motor.lib` (since there's no generic driver lib) and `interface_spi.lib`
- TPS2*: `power_management.lib`
- TPS5*/TPS6*: `power_switching_regulator.lib`
- TS5A*: `analog_switch.lib`
- TUSB*: `interface_usb.lib`
- TXB*: `logic_translator.lib`
- UCC289*: `power_pwm_controller.lib`

This is all that I see that needs to be mapped from the old to the new library. Please point out if you see any unmapped libraries or if you find parts which currently don't have a nice home.

A few more things:
- I believe all the libraries in the "The following libraries are yet to be assigned." from the first post can re removed.
- If we do what I advocate above, for the power management ICs, then perhaps the existing `power.lib` should be renamed to `power_symbols.lib` or `plumbing_symbols.lib`?
- I'm willing to clean up 
- I think we need some kind of framework to classify transformers. I mentioned this somewhere else because I've tried a few different ways at work without great success. Maybe someone has an idea? I'll try to lay out my past approaches later.

**Comment by SchrodingersGat** [comment]
@evanshultz I have made your proposed changes to the power libs, they make sense.

`power` could be renamed to `power_symbols`, this makes the most sense.

**Comment by keruseykaryu** [comment]
`power.lib` is a special Eeschema library. Its name can't be changed because this name is hardcoded in application. Remeber that, there are two component place command: first for devices, second for power ports. The second one uses `power.lib` as an exclusive source of symbols.

Going too far in libraries rearrangement you are lost some Eeschema basics!

**Comment by SchrodingersGat** [comment]
@keruseykaryu that is very good to point out. It was my understanding that the **Place Power** command searched all libraries and only displayed components marked as power?

**Comment by keruseykaryu** [comment]
Well. It seems we both have right. 

The old Eeschema (pre 4.0) searches for `power.lib` and display only components in this library when **Place power** command was used. New Eeschema indeed searches for power symbols in used libraries and allow to select any power port from any library.

So. If you need to be compatible with KiCad <4.0 you should stay with `power.lib`. For KiCad >4.0 it really doesn't matter, because new component selector code is more flexible.

**Comment by evanshultz** [comment]
Thanks @SchrodingersGat and @keruseykaryu. I didn't realize the special nature of `power.lib`.

Two things:
- `power_protection` is missing the `.lib` part
- `power_pfc.lib` should be `power_pfc_controller.lib`

**Comment by seppestas** [comment]
If we move the leds out of device.lib, wouldn't it make sense to rename device.lib to passive.lib indicating it contains passive components? Device is a pretty generic term imho.

**Comment by seppestas** [comment]
Wouldn't the sensors placed in sensor_multi-function.lib also fit in the sensor.lib library?

**Comment by poeschlr** [comment]
@seppestas wrote:
> If we move the leds out of device.lib, wouldn't it make sense to rename device.lib to passive.lib indicating it contains passive components? Device is a pretty generic term imho.

The generic led symbol is not removed from device lib.
The led lib will hold specialized symbols. (An example would be an atomic symbols for one special led that has the footprint pre assigned.)

**Comment by suzizecat** [comment]
> Maybe device.lib is not the best name but having a passive lib would not make it better. (some passive devices will be in a specialized lib with a footprint assigned)

Maybe something like `generic_device.lib` would be better ?

**Comment by seppestas** [comment]
> Maybe device.lib is not the best name but having a passive lib would not make it better. (some passive devices will be in a specialized lib with a footprint assigned)

What about the transformer.lib and sensors.lib both containing more generic components compared to the more specialized libs ("General purpose transformer" and "Sensors that do not fit other categories" respectively). Wouldn't having a library called passive.lib that contains "Generic / General purpose passive components" be consistent with this?

Of course doing this only makes sense if passive.lib only contains passive components, so there should not be leds in there. Personally I think having the generic led symbol in led.lib makes a lot more sense than having it in device.lib.

**Comment by stevenj** [comment]
> Of course doing this only makes sense if passive.lib only contains passive components, so there should not be leds in there. Personally I think having the generic led symbol in led.lib makes a lot more sense than having it in device.lib.

As a standard diode is a Passive component, arguably a LED is also a passive component, because electrically it behaves the same as a not light emitting diode.  Just Saying.

**Comment by suzizecat** [comment]
> As a standard diode is a Passive component, arguably a LED is also a passive component, because electrically it behaves the same as a not light emitting diode. Just Saying.

Is a standard diode even considered as a passive component ? (since on Farnell website, there is the "Discrete Semiconductor" (with diodes) and "Passive Components" with capacitors, resistors, etc)

**Comment by poeschlr** [comment]
I think currently the device lib is the only lib that by design contains only generic symbols. All other libs do not really contain general symbols but atomic or quasi atomic symbols. (or better: the goal for all other libs is to be atomic. Old symbols might not yet be atomic.)

The device lib also has transistors in there which are definitely not passive.
It has generic photo diodes which are not passive either. (specialized photo diodes are currently in opto.lib will be in sensor_photo in the new scheme.)
And a lot of other stuff.
The only general symbols in the current sensor.lib are the pt1000 and pt100 symbols. We could move them to the generic lib.
The opto lib does not contain such generic symbols. (feel free to check other libs. But i think this will be an ongoing process.)

I would be ok with splitting the generic lib. But it makes our live easier if we have clear libs where very generic symbols go. It makes it easy to define the look and feel of symbols in just one lib (or a specific set of libs) and tell everyone to use symbols that look similar to that lib.

**Comment by seppestas** [comment]
> As a standard diode is a Passive component, arguably a LED is also a passive component, because electrically it behaves the same as a not light emitting diode. Just Saying.

> Is a standard diode even considered as a passive component ? (since on Farnell website, there is the "Discrete Semiconductor" (with diodes) and "Passive Components" with capacitors, resistors, etc)

[According to Wikipedia](https://en.wikipedia.org/wiki/Electronic_component#Semiconductors) an LED (and all other type of diodes and semiconductors) is as active well, however I don't have a reference to a more authoritative source. My rule of thumb: if the function on the current and/or voltage is linear, it's passive. This of course only holds true for ideal devices.

**Comment by SchrodingersGat** [comment]
@poeschlr I agree with your direction for `device.lib` - independent of whether we keep this name or decide on something else, this is exactly the purpose of this library.

**Comment by suzizecat** [comment]
> According to Wikipedia an LED (and all other type of diodes and semiconductors) is as active well, however I don't have a reference to a more authoritative source. My rule of thumb: if the function on the current and/or voltage is linear, it's passive. This of course only holds true for ideal devices.

Well, in this case, a capacitor and an inductance (choke ?) aren't passive since the only component which i know to have a linear function between voltage and current is the resistor (U=RI) (capacitor and inductance got differential terms) 

According to the same good old [Wikipedia](https://en.wikipedia.org/wiki/Electronic_component#Passive_components)

>Components incapable of controlling current by means of another electrical signal are called passive devices. Resistors, capacitors, inductors, and transformers are all considered passive devices.

In this case it's a little bit tricky with the diode, but i guess the point is that the behavior of the LED isn't the same if the current flow through  anode-cathode direction or cathode-anode direction...

**Comment by evanshultz** [comment]
One other thing to mention is that Infineon makes former International Rectifier parts like [IRS2092](https://www.infineon.com/dgdl/irs2092.pdf?fileId=5546d462533600a401535675f1be2790). This is a gate driver, yes, but it's specifically intended for audio applications. Is it best to put this kind of part with other gate drivers, or in `audio_misc.lib`?

**Comment by poeschlr** [comment]
Back to the device lib discussion. (sorry)

We could name the current device lib something like basic_device. (would make it clearer what to expect in this lib.)

---

I noticed that the switches lib currently only holds generic switch symbols. (It seems in the current master all switches finally moved from device into switches lib. Therefore the latest patch for 4.x that adds this lib by default is really needed)
This means the switches and connectors lib currently only hold generic devices. (I would argue the logic family libs are also generic)

---

To clearly mark which libs hold generic devices we could also add generic as prefix/suffix.
This might make it easier to write the klc for the new naming convention and use of default footprint field.

We could then simply state that all symbols added to any lib that is not marked as generic needs the footprint field assigned and they also need to be named by the manufacturer part number. (or mpn base for devices produced by different manufacturers using different conventions for option fields.)
We could then even use travis to help checking this. (simply have a rule that is deactivated for libs having the generic prefix/suffix.)

Yes this would deviate from splitting the lib purely by function but a clearer KLC might be worth this tradeoff.

**Comment by evanshultz** [comment]
`transistors.lib` is massive. Almost 6000 lines as of now. Would it be good to split this up? I would think we would want to keep parts with the same symbol together so we can use ALIAS and for better maintenance, so perhaps we split by transistor type. `transistor_fet.lib`, `transistor_bipolar.lib` or `transistor_bjt.lib`, `transistor_igbt.lib`, etc. What do you think?

**Comment by evanshultz** [comment]
The transistor library could even be split more than I mentioned above, if desired. We could separate single and multiple transistor parts. This could result in `transistor_fet.lib` and `transistor_fet_array.lib`, or something. The core symbols would need to stay in sync fir visual purposes, but then it wouldn't be such a maintenance issue since the final symbols would still be unique in each library. Just an idea.

**Comment by jcassette** [comment]
In this repository, the connector symbol library has been renamed from `conn` to `Connector` (refer to first post)
However in [kicad-footprints](https://github.com/KiCad/kicad-footprints) the libraries have been renamed from `Connectors_*.pretty` to to `Conn_*.pretty`
What is the logic behind this?.. I think it should be better to keep the same name everywhere.
